src/components/footer.tsx

```diff
@@ -0,0 +1,84 @@
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import * as todosService from '../api/todos';
+
+type Props = {
+  onFilter: (filter: string) => void;
+  onError: () => void;
+  onTodos: () => void;
+  todos: Todo[];
+  filter: string;
+};
+
+export const Footer: React.FC<Props> = ({
+  onFilter,
+  onError,
+  onTodos,
+  todos,
+  filter,
+}) => {
+  function clearCompleted() {
+    todos.map(todo => {
+      if (todo.completed) {
+        todosService
+          .deleteTodo(todo.id)
+          .catch(() => onError('Unable to delete todo'));
+      }
+    });
+    onTodos(todos.filter(todo => !todo.completed));
+  }
+
+  enum Status {
+    All = 'all',
+    Active = 'active',
+    Completed = 'completed',
+  }
+
+  return (
+    <footer
+      className={classNames('todoapp__footer', {
+        hidden: todos.length === 0,
+      })}
+      data-cy="Footer"
+    >
+      <span className="todo-count" data-cy="TodosCounter">
+        {todos.filter(todo => !todo.completed).length} items left
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        {Object.values(Status).map((status, index) => {
```

Comment:
Don't need to use index here
```suggestion
        {Object.values(Status).map((status) => {
```